vllm/hpu/ops.py

```diff
@@ -148,3 +117,56 @@ def prompt_attention(
         attn_weights = attn_weights.flatten(1, 2)
     attn_weights = attn_weights.transpose(1, 2)
     return attn_weights
+
+
+class MoeMatmul(torch.nn.Module):
```

Comment:
How do you benefit it being a `torch.nn.Module` implementation?

Reply:
That way we can patch it in INC for quantization